src/Razor/src/Microsoft.CodeAnalysis.Razor/ComponentTagHelperDescriptorProvider.cs

```diff
@@ -81,7 +88,26 @@ private Compilation WithMetadataImportOptionsAll(Compilation compilation)
             return compilation.WithOptions(newCompilationOptions);
         }
 
-        private TagHelperDescriptor CreateDescriptor(ComponentSymbols symbols, INamedTypeSymbol type)
+        private TagHelperDescriptor CreateShortNameMatchingDescriptor(ComponentSymbols symbols, INamedTypeSymbol type)
+        {
+            var builder = CreateDescriptorBuilder(symbols, type);
+            builder.TagMatchingRule(r => r.TagName = type.Name);
+
+            return builder.Build();
+        }
+
+        private TagHelperDescriptor CreateFullyQualifiedNameMatchingDescriptor(ComponentSymbols symbols, INamedTypeSymbol type)
+        {
+            var builder = CreateDescriptorBuilder(symbols, type);
+            var containingNamespace = type.ContainingNamespace.ToDisplayString();
+            var fullName = $"{containingNamespace}.{type.Name}";
```

Comment:
How does this handle generic typed [components?](https://docs.microsoft.com/en-us/aspnet/core/razor-components/components?view=aspnetcore-3.0#generic-typed-components).  Or does it not get called on that path?

Reply:
Generic typed components will have typeparam set as a separate attribute just like in the example you linked. So the `type.Name` here would just be `ListViewTemplate` according to your example.